src/App.js

```diff
@@ -1,24 +1,46 @@
-import logo from './logo.svg';
-import './App.css';
+import React from "react";
+import { useState } from "react";
+import Navbar from "./compoents/Navbar";
+import Main from "./compoents/Main";
+import FavouriteAdd from "./compoents/FavouriteAdd";
+import PersonalFilmInfo from "./compoents/PersonalFilmInfo";
+import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
 
 function App() {
+  const [value, setValue] = useState("");
+  const [films, setFilms] = useState([]);
+  const [favourites, setFavourites] = useState([]);
+  const [genres, setGenres] = useState([])
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+      <Router>
+        <Switch>
+          <Route exact path="/favouriteadd">
+            <FavouriteAdd
+              films={films}
+              favourites={favourites}
+              setFavourites={setFavourites}
+            />
+          </Route>
+          <Route exact path="/personalfilminfo/:id">
+            <PersonalFilmInfo />
+          </Route>
+          <Route exact path="/">
+            <Navbar setValue={setValue} />
+            <Main
+              setGenres={setGenres}
+              genres={genres}
+              value={value}
+              films={films}
+              setFilms={setFilms}
+              setFavourites={setFavourites}
```

Comment:
You pass some props from state here only in the Main component. So maybe it's better to put those state variables in the Main component?